ModularTegustation/tegu_items/gadgets/unpowered.dm

```diff
@@ -100,29 +104,30 @@
 /obj/item/commandprojector/afterattack(atom/target, mob/user, proximity_flag)
 	. = ..()
 	if(cooldown <= world.time)
-		for(var/obj/effect/temp_visual/V in range(get_turf(target), 0))
-			if(is_type_in_typecache(V, commandtypes))
-				qdel(V)
-				return
-		switch(commandtype)
-			if(1)
-				new /obj/effect/temp_visual/commandMove(get_turf(target))
-			if(2)
-				new /obj/effect/temp_visual/commandWarn(get_turf(target))
-			if(3)
-				new /obj/effect/temp_visual/commandGaurd(get_turf(target))
-			if(4)
-				new /obj/effect/temp_visual/commandHeal(get_turf(target))
-			if(5)
-				new /obj/effect/temp_visual/commandFightA(get_turf(target))
-			if(6)
-				new /obj/effect/temp_visual/commandFightB(get_turf(target))
-			else
-				to_chat(user, "<span class='warning'>CALIBRATION ERROR.</span>")
+		for(var/obj/effect/temp_visual/HoloCommand/V in range(get_turf(target), 0))
+			qdel(V)
+			return
+		if(!CheckHoloCapacity())
+			to_chat(user, "<span class='warning'>COMMAND CAPACITY REACHED.</span>")
+			return
+		if(commandtype > 0 && commandtype <= 6)
+			var/thing_to_spawn = commandtypes[commandtype]
+			var/thing_spawned = new thing_to_spawn(get_turf(target))
+			current_commands += thing_spawned
+		else
+			to_chat(user, "<span class='warning'>CALIBRATION ERROR.</span>")
 		cooldown = world.time + commanddelay
 	playsound(src, 'sound/machines/pda_button1.ogg', 20, TRUE)
 
-
+//Checks if we had exceeded our max amount of commands, checks if each command is real.
+/obj/item/commandprojector/proc/CheckHoloCapacity()
+	if(current_commands.len > max_commands)
+		for(var/obj/effect/temp_visual/HoloCommand/i in current_commands)
+			if(QDELETED(i))
```

Comment:
You should consider using a signal instead of this.

Reply:
I checked signals and theres no signals for a atom being deleted.

Reply:
There is a signal for it, yes.

Reply:
This?
/// before a datum's Destroy() is called: (force), returning a nonzero value will cancel the qdel operation
#define COMSIG_PARENT_PREQDELETED "parent_preqdeleted"
/// just before a datum's Destroy() is called: (force), at this point none of the other components chose to interrupt qdel and Destroy will be called
#define COMSIG_PARENT_QDELETING "parent_qdeleting"

Reply:
COMSIG_PARENT_QDELETING apparently works. Ill be honest i doubted you egor but this does make things more simple. I just dont understand why it says it only effects datums.

Reply:
Datum is the bottom level object. EVERYTHING is a datum(might be wrong here, not sure if areas and turfs are a datum).